processor/batchprocessor/batch_processor.go

```diff
@@ -274,3 +287,44 @@ func (bm *batchMetrics) add(item interface{}) {
 	bm.metricCount += uint32(newMetricsCount)
 	md.ResourceMetrics().MoveAndAppendTo(bm.metricData.ResourceMetrics())
 }
+
+type batchLogs struct {
+	nextConsumer consumer.LogsConsumer
+	logData      pdata.Logs
+	logCount     uint32
+}
+
+func newBatchLogs(nextConsumer consumer.LogsConsumer) *batchLogs {
+	b := &batchLogs{nextConsumer: nextConsumer}
+	b.reset()
+	return b
+}
+
+func (bm *batchLogs) export(ctx context.Context) error {
+	return bm.nextConsumer.ConsumeLogs(ctx, bm.logData)
+}
+
+func (bm *batchLogs) itemCount() uint32 {
+	return bm.logCount
+}
+
+func (bm *batchLogs) size() int {
+	return bm.logData.ResourceLogs().Len()
```

Comment:
What is size() supposed to return?

Reply:
Based on the metrics batcher, the size in bytes of the internal data representation of logs.  I added a helper method to pdata to derive it without the batch processor having to convert to internal.

Reply:
Do we have batching by size in bytes? I thought we only support batching by the number of items. Can you point me to the code which does byte-based batching? I don't see anything mentioned in the config of the batcher.

Reply:
It is just used for telemetry at the detailed level to record how many bytes the batches are.